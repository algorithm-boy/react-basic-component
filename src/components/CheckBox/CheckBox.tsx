import React, { SFC } from 'react';
import { GeneralProps } from '../global.d';

export interface CheckBoxProps extends GeneralProps {
  id?: string | number;
  name? : number | string;
  disabled?: boolean;
  defaultChecked?: boolean;
  className? : string;
  style? : React.CSSProperties;
  checked?: boolean;
  readOnly?: boolean;
  onChange? : (event?: any) => void;
};

const CheckBox: SFC<CheckBoxProps> = props => {
  const { className, style = {}, children, disabled } = props;
  return (
    <div
      className={className}
      style={style}
    >
      <p>
        <input 
          type="checkbox" 
          disabled={disabled || props.readOnly}
          defaultChecked={props.defaultChecked}
          className={props.className}
          style={props.style}
          checked={props.checked}
          readOnly={props.readOnly}
          onChange={(e)=>{
            props.onChange?props.onChange(e.target.checked):'';
          }}  
        />
        {children}
      </p>
    </div>
  );
};

export default CheckBox;
