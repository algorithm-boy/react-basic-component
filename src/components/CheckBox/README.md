# CheckBox

## API

| 参数(props) | 说明(specification) | 类型(type) | 默认值(defaultProps) |
| --- | --- | --- | --- |
| className(optional) | CheckBox的类名 (The class property for CheckBox) | string | - |
| style(optional) | CheckBox的行内样式 (The inline style for CheckBox) | object | {} |

## Example

```javascript
<CheckBox
  className={'hfdc-CheckBox'}
  style={{
    width: '360px'
  }}
/>
```

