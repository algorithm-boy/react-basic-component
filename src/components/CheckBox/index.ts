import CheckBox from './CheckBox';
export { CheckBoxProps } from './CheckBox';
export default CheckBox;
