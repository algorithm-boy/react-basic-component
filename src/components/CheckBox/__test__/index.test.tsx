import * as React from 'react';
import { configure, shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import CheckBox from '../index';

configure({ adapter: new Adapter() });

describe('CheckBox', () => {
  it('renders correctly', () => {
    const wrapper = render(
      <CheckBox>Test</CheckBox>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
