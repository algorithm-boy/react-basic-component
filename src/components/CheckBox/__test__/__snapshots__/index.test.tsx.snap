// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "attribs": Object {
              "type": "checkbox",
            },
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "data": "Test",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "text",
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "type": undefined,
            },
            "x-attribsPrefix": Object {
              "type": undefined,
            },
          },
          Object {
            "data": "Test",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "type": "checkbox",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "type": undefined,
              },
              "x-attribsPrefix": Object {
                "type": undefined,
              },
            },
            "type": "text",
          },
        ],
        "name": "p",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
